src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointOptions.tsx

```diff
@@ -17,25 +18,38 @@ function BreakpointLineContents({ breakpoint }: BreakpointProps) {
   const replayClient = useContext(ReplayClientContext);
   const { sourceId } = breakpoint.location;
 
-  const sourceLines = getSourceLinesSuspense(sourceId, replayClient);
+  const streamingSourceContent = getStreamingSourceContentsSuspense(replayClient, sourceId);
+  const parsed = getParsedValueIfCached(streamingSourceContent, null)?.value;
 
-  const snippet = useMemo(() => {
+  const parsedTokensByLine = parsed?.parsedTokensByLine;
+  const rawTextByLine = parsed?.rawTextByLine;
+
+  const { snippet, tokens } = useMemo((): { snippet: string; tokens: ParsedToken[] } => {
     const { column, line } = breakpoint.location;
     let snippet = "";
+    let tokens: ParsedToken[] = [];
 
-    if (sourceLines.length > 0) {
-      const lineText = sourceLines[line - 1];
-      return lineText.slice(column, column! + 100).trim();
+    if (
+      rawTextByLine &&
+      rawTextByLine.length > 0 &&
+      parsedTokensByLine &&
+      parsedTokensByLine[line - 1]
+    ) {
+      const lineText = rawTextByLine[line - 1];
+      tokens = parsedTokensByLine[line - 1].slice(
+        parsedTokensByLine[line - 1].findIndex(parsedToken => parsedToken.columnIndex === column)
+      );
+      return { snippet: lineText.slice(column, column! + 100).trim(), tokens };
```

Comment:
I think these two should be checked separately?

```suggestion
const { snippet, tokens } = useMemo((): {
  snippet: string;
  tokens: ParsedToken[];
} => {
  const { column, line } = breakpoint.location;

  let snippet: string | null = null;
  let tokens: ParsedToken[] | null = null;

  if (rawTextByLine && rawTextByLine.length > 0) {
    const line = rawTextByLine[line - 1];
    snippet = line.slice(column, column! + 100).trim();
  }

  if (parsedTokensByLine && parsedTokensByLine[line - 1]) {
    const line = parsedTokensByLine[line - 1];
    tokens = line.slice(
      line.findIndex((parsedToken) => parsedToken.columnIndex === column)
    );
  }

  return { snippet, tokens };
}, [rawTextByLine, parsedTokensByLine, breakpoint]);
```

It's possible we will have loaded the plaintext source code without parsing it.